init/eessi_archdetect.sh

```diff
@@ -150,8 +150,25 @@ cpupath(){
     fi
 }
 
+accelpath() {
+    # If EESSI_ACCELERATOR_TARGET_OVERRIDE is set, use it
+    log "DEBUG" "accelpath: Override variable set as '$EESSI_ACCELERATOR_TARGET_OVERRIDE' "
+    [ $EESSI_ACCELERATOR_TARGET_OVERRIDE ] && echo ${EESSI_ACCELERATOR_TARGET_OVERRIDE} && exit
+
+    # check for NVIDIA GPUs via nvidia-smi command
+    nvidia_smi=$(command -v nvidia-smi)
+    if [[ $? -eq 0 ]]; then
+        log "DEBUG" "accelpath: nvidia-smi command found @ ${nvidia_smi}"
+	gpu_info=$(nvidia-smi --query-gpu=gpu_name,count,driver_version,compute_cap --format=csv,noheader | head -1)
```

Comment:
`nvidia-smi` can exist but fail to run (for example you can have the drivers installed on a node that does not have an actual GPU). In the driver script we implement:
https://github.com/EESSI/software-layer/blob/720787c0c3ce4e5c606b745f5a578761827b9a04/scripts/gpu_support/nvidia/link_nvidia_host_libraries.sh#L48-L58

Reply:
OK, I'll rework this to check exit code of `nvidia-smi` command as well.

Maybe even make a small utility function for it that we can use in both places?

Reply:
Yes, a reusable utility function is a good idea, would also allow us to adopt to changes in the output of `nvidia-smi` if they happen in the future

Reply:
Adding a reusable function to call `nvidia-smi` to `scripts/utils.sh` would make archdetect dependent on it, which is maybe not a good idea, so I didn't take this route...

It's also a bit tricky & pointless anyway, since in `link_nvidia_host_libraries.sh` there are two calls to `nvidia-smi` which can not be combined into one...

Reply:
I may switch what we have in `link_nvidia_host_libraries.sh` to use the single command that you have to extract both bits of information